### PR TITLE
Add handling for FormatException thrown by invalid feature flag json

### PR DESCRIPTION
**Pull request by amerjusupovic**

### CODE REVIEW

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
         public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, Logger logger, CancellationToken cancellationToken)
         {
-            FeatureFlag featureFlag = ParseFeatureFlag(setting.Key, setting.Value);
+            FeatureFlag featureFlag = ParseFeatureFlag($"Failed to parse invalid feature flag json. Key: '{setting.Key}'", setting.Value);
```
Comment (avanigupta) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs:25]: Is this intentional?

Comment (amerjusupovic) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs:25]: This looks weird now that I see it again, the reason I did it was `Setting.key` was only being used for error logging but it should definitely be a different variable name or a class variable.

Comment (avanigupta) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs:25]: Let's pass the key and log the message wherever its needed. I thought this was accidentally copy/pasted :)

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Constants/LoggingConstants.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@ internal class LoggingConstants
         public const string RefreshFailedDueToKeyVaultError = "A refresh operation failed while resolving a Key Vault reference.";
         public const string PushNotificationUnregisteredEndpoint = "Ignoring the push notification received for the unregistered endpoint";
         public const string FallbackClientLookupError = "Failed to perform fallback client lookup.";
+        public const string RefreshFormattingError = "A refresh operation failed due to a formatting error.";
```
Comment (avanigupta) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Constants/LoggingConstants.cs:18]: `RefreshFailedDueToFormattingError` matches existing naming pattern.

**Review by avanigupta — APPROVED** [review]

**Pull request merged**